auth0/src/main/scala/io/unsecurity/auth/auth0/oidc/Auth0OidcSecurityContext.scala

```diff
@@ -162,7 +162,7 @@ class Auth0OidcSecurityContext[F[_]: Sync, U](val authConfig: AuthConfig,
   }
 
   def fetchTokenFromAuth0(code: String, auth0CallbackUrl: URI): Directive[F, TokenResponse] = {
-    val tokenUrl      = Uri.unsafeFromString(s"https://${authConfig.authDomain}/oauth/token")
+    val tokenUrl      = Uri.unsafeFromString(authConfig.issuer) / "oauth" / "token"
```

Comment:
should not authConfig.issuer then just be a URI instead of a string ?

Reply:
The answer is "it depends". Let's take a look at the verifier from Auth0 that we build in `TokenVerifier.scala`.

```scala
val verifier = JWT
  .require(alg)
  .withIssuer(issuer)
  .withAudience(clientId)
  .build()
```

Here, `issuer` is a string and it is used to perform an exact comparison with the `iss` property of the JWT. I say exact, because it's not enough that the URLs are seemingly functionally identical. If a JWT comes with `iss` equal `http://foo.com/` and we provide to the verifier `http://foo.com` (notice the lacking /), then the verification will fail.

Now, I've confirmed that

```scala
Uri.unsafeFromString("http://foo.com/").toString # => "http://foo.com/"
Uri.unsafeFromString("http://foo.com").toString # => "http://foo.com"
```

... IE. `Uri` performs no normalization in these cases and `Uri.unsafeFromString(X).toString == X` holds. Is this predicate however guaranteed? If it is, then it is safe to require an Uri instead of String. If this however is not the case, then the normalization will possibly disallow a developer to consume some providers with certain URLs.